core/trino-main/src/main/java/io/trino/metadata/SystemSecurityMetadata.java

```diff
@@ -184,6 +184,16 @@ default void validateEntityKindAndPrivileges(Session session, String entityKind,
      */
     Optional<Identity> getFunctionRunAsIdentity(Session session, CatalogSchemaFunctionName functionName);
 
+    /**
+     * Set the owner of the specified function
+     */
+    void setFunctionOwner(Session session, CatalogSchemaFunctionName function, TrinoPrincipal principal);
+
+    /**
+     * A function is created
+     */
+    void functionCreated(Session session, CatalogSchemaFunctionName function, TrinoPrincipal principal);
```

Comment:
What's `principal` for? Isn't `session` enough to get the identity of the creator of the function?

Reply:
following convention from other methods

Reply:
ok i got it, fixed, thanks